src/storage/data_table.cpp

```diff
@@ -593,13 +775,13 @@ bool DataTable::CheckForeignKeyConstraints(
       if (index->GetIndexType() == IndexConstraintType::PRIMARY_KEY) {
         LOG_INFO("BEGIN checking referred table");
 
-        std::vector<oid_t> key_attrs = foreign_key->GetSourceColumnIds();
+        std::vector<oid_t> key_attrs = foreign_key->GetSinkColumnIds();
         std::unique_ptr<catalog::Schema> foreign_key_schema(
-            catalog::Schema::CopySchema(schema, key_attrs));
+            catalog::Schema::CopySchema(ref_table->schema, key_attrs));
         std::unique_ptr<storage::Tuple> key(
             new storage::Tuple(foreign_key_schema.get(), true));
         // FIXME: what is the 3rd arg should be?
-        key->SetFromTuple(tuple, key_attrs, index->GetPool());
+        key->SetFromTuple(tuple, foreign_key->GetSourceColumnIds(), index->GetPool());
```

Comment:
I know 3rd argument is unchanged... re: the FIXME comment, is there still a question about the 3rd argument?